src/components/portal/portal.tsx

```diff
@@ -172,10 +177,11 @@ export class Portal {
         }
 
         if (this.inheritParentWidth) {
-            this.container.style.width =
-                rect.width > 0
-                    ? `${rect.width}px`
-                    : `${this.getContentWidth(this.container)}px`;
+            const containerWidth = this.getContentWidth(this.container);
+            this.container.style.width = `${containerWidth}px`;
+            if (rect.width > 0) {
+                this.container.style.width = `${rect.width}px`;
+            }
```

Comment:
Hmm… I think this might trigger two DOM accesses if `rect.width > 0` is true…

I'd like to split out the addition of typings to a separate commit, and also rename that bloody  `rect` variable, so I'll add my suggestion in a fixup.